src/main/java/ai/knowly/langtorch/store/vectordb/PGVectorStore.java

```diff
@@ -0,0 +1,342 @@
+                            similaritySearchQuery.getTopK()
+                    )
+            );
+
+            neighborStmt.setObject(1, new PGvector(queryVectorValuesAsFloats));
+            ResultSet result = neighborStmt.executeQuery();
+
+
+            while (result.next()) {
+                String vectorId = (String) result.getObject(1);
+                PGvector pGvector = (PGvector) result.getObject(2);
+                String key = (String) result.getObject(3);
+                String value = (String) result.getObject(4);
+
+                double[] currentVector = getDoubleVectorValues(pGvector.toArray());
+
+                double score = distanceStrategy.calculateDistance(queryVectorValuesAsDoubles, currentVector);
+
+                documentsWithScoresMap.computeIfAbsent(vectorId, s -> {
+                    Metadata defaultMetadata = Metadata.builder().build();
+                    return DomainDocument.builder()
+                            .setId(vectorId)
+                            .setPageContent("")
+                            .setSimilarityScore(Optional.of(score))
+                            .setMetadata(defaultMetadata)
+                            .build();
+                });
+
+                DomainDocument documentWithScore = documentsWithScoresMap.get(vectorId);
+                saveValueToMetadataIfPresent(documentWithScore, key, value);
+                documentsWithScoresMap.put(vectorId, getDocumentWithScoreWithPageContent(documentWithScore, key, value));
+            }
+            documentsWithScores = new ArrayList<>(documentsWithScoresMap.values());
+        } catch (SQLException e) {
+            logger.atSevere().withCause(e).log("Error with SQL Exception");
+            return new ArrayList<>(documentsWithScoresMap.values());
+        }
+
+        return documentsWithScores;
+    }
+
+    private void createEmbeddingsTable() throws SQLException {
+        pgVectorService.executeUpdate(
+                sqlCommandProvider.getCreateEmbeddingsTableQuery(pgVectorStoreSpec.getVectorDimensions())
+        );
+    }
+
+    private void createMetadataTable() throws SQLException {
+        pgVectorService.executeUpdate(
+                sqlCommandProvider.getCreateMetadataTableQuery()
+        );
+    }
+
+    private PGVectorQueryParameters getVectorQueryParameters(List<DomainDocument> documents) {
+        List<PGVectorValues> vectorValues = new ArrayList<>();
+        StringBuilder vectorParameters = new StringBuilder();
+        StringBuilder metadataParameters = new StringBuilder();
+        int metadataSize = 0;
+        for (DomainDocument document : documents) {
+            List<Double> vector = createVector(document);
+            String id = document.getId().orElse(UUID.randomUUID().toString());
+
+            PGVectorValues pgVectorValues = buildPGVectorValues(id, vector, document.getMetadata());
+            vectorValues.add(pgVectorValues);
```

Comment:
merge into one line?
buildPGVectorValues(id, vector, document.getMetadata()).add(pgVectorValues);

as pgVectorValues is not used